CONTRIBUTING.md

```diff
@@ -501,22 +501,25 @@ Before merging a pull request:
 
 ### Git Commit Style
 
-We follow the [Go style guide on commit messages][go-git-commit-style]. Write
-concise commits that start with the package name and have a description that
-finishes the sentence "This change modifies CometBFT to...". For example,
+We follow the [Conventional Commits][conventional-commits] spec. Write concise
+commits that start with a type (`fix`, `feat`, `chore`, `ci`, `docs`, etc.) and
+an optional scope - package name (e.g., `feat(internal/consensus)`), followed
+by a description that finishes the sentence "This change modifies CometBFT
+to...".
+
+If the commit introduces a breaking change, append the `!` after the scope
+(e.g., `feat(internal/consensus)!`).
+
+For example,
 
 ```sh
-cmd/debug: execute p.Signal only when p is not nil
+fix(cmd/cometbft/commands/debug): execute p.Signal only when p is not nil
 
 [potentially longer description in the body]
 
 Fixes #nnnn
 ```
 
-Each PR should have one commit once it lands on `main`; this can be accomplished
-by using the "squash and merge" button on GitHub. Be sure to edit your commit
-message, though!
-
```

Comment:
I think this still makes sense (except for PR that represent a merge from `main` into a feature branch)

Reply:
This concerns me as well: does this mean we change from squash-and-merge to preserving commit history in PRs, and therefore imposing more discipline on developers' branches? I'm not a fan of squash-and-merge myself, but I feel this needs some buy-in from the team.

This `!` thing may be particularly vexing: do we want to be forced to think of which changes might be breaking, when making every single commit (or, perhaps, when grooming commits pre-merge)?

Reply:
> does this mean we change from squash-and-merge to preserving commit history in PRs,

no. I removed this paragraph because we enforce it via Github settings, so I felt that there's no real need to state that **if there's no other way to merge your PR other than squash merge**. I can revert of course.

> This ! thing may be particularly vexing: do we want to be forced to think of which changes might be breaking, when making every single commit (or, perhaps, when grooming commits pre-merge)?

no

Reply:
It would help to continue keeping this remark about squash and merge in the CONTRIBUTING.md doc.